rust/core-lib/Cargo.toml

```diff
@@ -11,6 +11,7 @@ serde = "1.0"
 serde_json = "1.0"
 serde_derive = "1.0"
 time = "0.1"
+parking_lot = "0.4"
```

Comment:
I'm actually going to ask this be a separate change, when I look at the dependencies it pulls in I think it's going to cause difficulty for Fuchsia. We should fix that, of course, but it feels pretty orthogonal to the thrust of this PR.